Cluster Programs

```diff
@@ -1,3 +1,68 @@
+Wednesday 07/25
+
+1) I have review Knitr and the reproducible data in coursera and given that it's implemented in Rstudio, 
+the process to use it should be pretty straight forward, the only problem is that we need to have the entire code running, 
+knitr will process the code and print the results.
+At the moment I haven't begin writing the movement to knitr because I'm still struggling with the optimization of the code.
+
+2) Documentation about the missing values in nclass was unavailable. 
+This variable comes in the original datasets from Harvard Dataverse, it is NOT created in the SAS code. 
+In my opinion, we could drop the NAs in the code, given that we are mainly grouping by nclass and NAs could mean that we are grouping 
```

Comment:
Please check and see how the equivalent of the NAs were handled in the SAS code so we can factor this into making a decision. Also, please scan the Thomson, Fox-Kean and Jaffe, Trajtenberg and Henderson (1993) papers to see if they mention what they do with missing values.

Reply:
By looking at the SAS code. 
Possiblenclass has 911 NAs and Matching2 (before its merge with all the if conditions) has 175 NAs. Once its merge, Matching 2 (running all the if conditions) ends up with 94 NAs. 
Now, here I find something strange, the first 16 matches are different, but the rest 78 is the same patent (same cited and patent) that it's been matched to different controls. 
In the next step of the SAS code, it apply nodupkey, dropping all this NA's. I end up with 17 NA's, after the first iteration. 
The code never deals with the NAs, not even in the third program. 

Regarding the literature, I found this quote in Thompson 2005  that summarize the matching for both papers. 
"JTH selected controls in which the primary class of the citing patent matched the primary class of its control patent. Thompson 2005, matched the primary class of the citing patent to any class of the selected control patent. (Page 7, Thompson, 2005). 

When reading and reviewing JTH paper, it makes no comment on the missing values. 
In my understanding, Thompson is matching the primary class with whatever class, this implied that he doesn't care if he is matching missing values with actual classes. 
In this paper, there is no comment on how he deals with missing values either.

Reply:
Not quite
When I re-run the SAS code and I got the same matchings, but I found something else. 

I'm going to write it as a pull request so its easy to explain and to comment.